### ISSUE TITLE
Can't instantiate abstract class _Interceptor with abstract method __await__

### ISSUE THREAD

**Issue by RockChinQ** [graingert/taskgroup#20]
Hello, I've a project deps on `Quart` and `Hypercorn` , when I build the latest image yesterday, I found the quart cannot work.  
After a hard check, I found it was caused by `taskgroup`'s upgrading. It works well with `taskgroup==0.0.0a4` but broken with `taskgroup==0.1.1`(also in 0.0.0a5). I thought it was because the new version `taskgroup` has some conflicts with my own task management. I'm trying to find it out, but this repository has much too little information, could you add a detailed README or other documentation for `taskgroup` ？Thank you.

**Comment by graingert** [comment]
I've provided a bit more info in https://github.com/graingert/taskgroup/pull/21

could you describe more about your issue? Does it fail in 3.11 or 3.12 in the same ways?

**Comment by graingert** [comment]
could you also try using anycorn instead of hypercorn and see if the issue persists?

**Comment by RockChinQ** [comment]
Thank you so much for replying and for the PR. I just locked the version to `0.0.0a4 ` to temporarily solve this issue, but I haven't found the deep cause yet. I'll reply to this issue as soon as I work it out.

**Comment by monoatomix** [comment]
I also encountered the same issue on python 3.9.x with following error

```
Traceback (most recent call last):
  File "..\lib\multiprocessing\process.py", line 315, in _bootstrap
    self.run()
  File "..\lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "..\lib\site-packages\hypercorn\asyncio\run.py", line 196, in asyncio_worker
    _run(
  File "..\lib\site-packages\hypercorn\asyncio\run.py", line 234, in _run
    runner.run(main(shutdown_trigger=shutdown_trigger))
  File "..\lib\site-packages\taskgroup\runners.py", line 127, in run
    task = _task_factory(self._loop, coro, context=context)
  File "..\lib\site-packages\taskgroup\tasks.py", line 89, in task_factory
    return Task(coro, loop=loop, **kwargs)
  File "..\lib\site-packages\taskgroup\tasks.py", line 63, in __init__
    coro = _Interceptor(coro, context)
TypeError: Can't instantiate abstract class _Interceptor with abstract method __await__
..\lib\multiprocessing\process.py:331: RuntimeWarning: coroutine 'worker_serve' was never awaited
  traceback.print_exc()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

Happening in both anycorn and hypercorn while running with more than 1 workers.
Python 3.11 works fine

I saw #23, hopefully that fixes it.

**Comment by graingert** [comment]
ah ok looks like an easy fix (that wasn't fixed in #23), I'll get on it.

**Comment by graingert** [comment]
@M0N0Atomic can you check if this fixes it?

https://github.com/graingert/taskgroup/pull/25

**Comment by monoatomix** [comment]
Yes that fixed it.

**Comment by graingert** [comment]
@M0N0Atomic thanks for the report, released in 0.2.1!
